base/tests/test_MAPL_ISO8601_Time.pf

```diff
@@ -0,0 +1,486 @@
+
+contains
+
+   @test
+   subroutine test_date_fields()
+      integer :: y
+      integer :: m
+      integer :: d
+      type(date_fields) :: df
+     
+      y = 2022
+      m = 06
+      d = 27
+     
+      df = date_fields(y, m, d)
+
+      @assertEqual(y, df % year)
+      @assertEqual(m, df % month)
+      @assertEqual(d, df % day)
+
+   end subroutine test_date_fields
+
+   @test
+   subroutine test_date_fields_custom_constructor()
+      type(date_fields) :: df
+      integer, dimension(3) :: f = [2022, 06, 27]
+     
+      df = date_fields(f)
+
+      @assertEqual(f(1), df % year)
+      @assertEqual(f(2), df % month)
+      @assertEqual(f(3), df % day)
+
+   end subroutine test_date_fields_custom_constructor
+
+   @test
+   subroutine test_time_fields()
+      integer :: h
+      integer :: m
+      integer :: s
+      integer :: ms
+      type(time_fields) :: tf
+     
+      h = 13
+      m = 02
+      s = 27
+      ms = 37
+     
+      tf = time_fields(h, m, s, ms)
+
+      @assertEqual(h, tf % hour)
+      @assertEqual(m, tf % minute)
+      @assertEqual(s, tf % second)
+      @assertEqual(ms, tf % millisecond)
+
+   end subroutine test_time_fields
+
+   @test
+   subroutine test_time_fields_custom_constructor()
+      type(time_fields) :: tf
+      integer, dimension(4) :: f = [13, 02, 27, 36]
+     
+      tf = time_fields(f)
+
+      @assertEqual(f(1), tf % hour)
+      @assertEqual(f(2), tf % minute)
+      @assertEqual(f(3), tf % second)
+      @assertEqual(f(4), tf % millisecond)
+
+   end subroutine test_time_fields_custom_constructor
+
+   @test
+   subroutine test_is_digit()
+      integer :: i = -1
+      integer, parameter :: imin = 0
+      integer, parameter :: imax = 127
+      integer, parameter :: i1 = iachar('0') - 1
+      integer, parameter :: i2 = iachar('9') + 1
+      integer, parameter :: acodes(*) = [(i, i = imin, i1), (i, i = i2, imax)]
+      integer :: i0 = iachar('0')
+
+      do i = 0, 9
+         @assertTrue(is_digit(achar(i0+i)))
+      end do
+
+      do i = 1, size(acodes)
+         @assertFalse(is_digit(achar(acodes(i))))
+      end do
+
+   end subroutine test_is_digit
+
+   @test
+   subroutine test_are_digits()
+      @assertTrue(are_digits('0123456789'))
+      @assertTrue(are_digits('0'))
+      @assertTrue(are_digits('9'))
+      @assertFalse(are_digits('ABCD'))
+      @assertFalse(are_digits('A234'))
+      @assertFalse(are_digits('1B34'))
+      @assertFalse(are_digits('123Z'))
+      @assertFalse(are_digits('AB3D'))
+   end subroutine test_are_digits
+
+   @test
+   subroutine test_make_char_array()
+      character(len=*), parameter :: char_string = 'ABCDEFGHIJ'
+      character, dimension(len(char_string)) :: char_array
+      integer :: i = -1
+
+      char_array = make_char_array(char_string)
+
+      @assertEqual(len(char_string), size(char_array))
+      
+      do i=1, len(char_string)
+         @assertEqual(char_string(i:i), char_array(i))
+      end do
```

Comment:
I think this can just be:
```suggestion     
         @assertEqual(char_string, char_array)
```
But it is possible that this rarely used overload is missing.